test/unit/view.child-views.spec.js

```diff
@@ -213,77 +213,88 @@ describe('layoutView', function() {
   });
 
   describe('when destroying', function() {
-    beforeEach(function() {
-      this.layoutViewManager = new this.View(this.viewOptions);
-      $('<span id="parent">').append(this.layoutViewManager.el);
-      this.layoutViewManager.render();
+    var oldLayoutView;
 
-      this.regionOne = this.layoutViewManager.regionOne;
-      this.regionTwo = this.layoutViewManager.regionTwo;
+    before(function() {
+      this.setup = function(viewOptions) {
+        this.layoutViewManager = new this.View(viewOptions);
+        $('<span id="parent">').append(this.layoutViewManager.el);
+        this.layoutViewManager.render();
 
-      var View = Marionette.View.extend({
-        template: false,
-        destroy: function() {
-          this.hadParent = this.$el.closest('#parent').length > 0;
-          return View.__super__.destroy.call(this);
-        }
-      });
+        this.regionOne = this.layoutViewManager.regionOne;
+        this.regionTwo = this.layoutViewManager.regionTwo;
 
-      this.regionOneView = new View();
-      this.regionOne.show(this.regionOneView);
+        var View = Marionette.View.extend({
+          template: false,
+          destroy: function() {
+            this.hadParent = this.$el.closest('#parent').length > 0;
+            return View.__super__.destroy.call(this);
+          }
+        });
+
+        this.regionOneView = new View();
+        this.regionOne.show(this.regionOneView);
 
-      this.sinon.spy(this.regionOne, 'empty');
-      this.sinon.spy(this.regionTwo, 'empty');
+        this.sinon.spy(this.regionOne, 'empty');
+        this.sinon.spy(this.regionTwo, 'empty');
 
-      this.sinon.spy(this.layoutViewManager, 'destroy');
-      this.layoutViewManager.destroy();
-      this.layoutViewManager.destroy();
+        this.sinon.spy(this.layoutViewManager, 'destroy');
+        this.layoutViewManager.destroy();
+        this.layoutViewManager.destroy();
+      };
     });
 
     it('should empty the region managers', function() {
+      this.setup();
```

Comment:
The main reason to called setup is because it needs to be configurable, beforeEach does not have that ability, the previous solution, which used beforeEach, relied on the order of the tests to set up the viewOptions for the beforeEach block.

The function setup was introduced to make the beforeEach block configurable, so now each test can configure the viewOptions for the test.

I don't know if I'm making my self clear here